### ISSUE TITLE
Working with Docker Machine

### ISSUE THREAD

**Issue by alexlafroscia** [artificialio/sane#162]
Is is possible to use Sane with Docker Machine?  I can't imagine why it wouldn't be, but I can't seem to get it to work.  I've got a Docker machine running:

```
NAME          ACTIVE   DRIVER         STATE     URL                         SWARM
giteasy-dev            virtualbox     Stopped                               
rancher-do             digitalocean   Stopped   tcp://45.55.156.139:2376    
sane          *        virtualbox     Running   tcp://192.168.99.100:2376   
```

and that Docker server has the image that it needs:

```
REPOSITORY                TAG                 IMAGE ID            CREATED             VIRTUAL SIZE
mongo                     latest              077c6501ee18        4 days ago          257.8 MB
artificial/docker-sails   stable-pm2          dd628713b131        10 weeks ago        304.9 MB
```

but whenever I try to run `sane up --docker`, I get the following error:

> Please start docker/boot2docker and run the command again.

Is it possible that I'm missing a step somewhere?  Thanks for the help!

**Comment by alexlafroscia** [comment]
I checked out the source for how Sane determines if Docker is running, and it doesn't seem to support Docker Machine. I am going to put a PR together to see if I can add support.

**Comment by alexlafroscia** [comment]
It seems as if running `npm install` after cloning the repository modifies some files. What's up with that?

**Comment by Globegitter** [comment]
@alexlafroscia yes, I haven't used machine at all yet, so anything that adds support for that is highly welcome. And what you are seeing with the `npm install` is a weird npm behaviour from `prepublish` here: https://github.com/artificialio/sane/blob/master/package.json#L9

**Comment by alexlafroscia** [comment]
Oh okay, so I should just ignore the changes that are being made there? Or commit the differences?

**Comment by Globegitter** [comment]
we are using babel to transpile ES7 code to ES5, so just change the file here: https://github.com/artificialio/sane/blob/master/src/tasks/checkEnvironment.js and run either `npm run build` or `npm run build:watch` to transpile the code.

**Comment by alexlafroscia** [comment]
Cool, thanks. It seems like the change will be a little more invasive that I thought initially; just checking for `docker-machine` isn't enough, since there's a whole bunch of code that assumes that if `boot2docker` exists then you must want to use that.  All of that would have to change to give precedence to `docker-machine` first.

**Comment by Globegitter** [comment]
@alexlafroscia could you give a quick summary on the differences between `boot2docker/docker-machine`? When would you use `docker-machine` over `boot2docker`?

**Comment by alexlafroscia** [comment]
Do basically, `docker-machine` is what Docker is moving toward to manage different Docker hosts across all platforms.  It uses `boot2docker` under the hood, but you don't interact with it directly anymore.  Instead, you use `docker-machine` to point the `docker` command at different hosts, either locally or remotely.
